cpu/cc2538/startup-gcc.c

```diff
@@ -30,22 +30,23 @@
  *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 /**
- * \addtogroup cc2538dk
+ * \addtogroup cc2538
  * @{
  *
  * \file
- * Startup code for the cc2538dk platform, to be used when building with gcc
+ * Startup code for the cc2538 chip, to be used when building with gcc
  */
 #include "contiki.h"
 #include "reg.h"
 #include "flash-cca.h"
 #include "sys-ctrl.h"
-#include "uart.h"
 
 #include <stdint.h>
 /*---------------------------------------------------------------------------*/
 extern int main(void);
 /*---------------------------------------------------------------------------*/
+#define WEAK_ALIAS(x) __attribute__ ((weak, alias(#x)))
```

Comment:
> SDCC, cc65.

These are indeed very different and do not support this feature.

> I think the extent of what you are suggesting would need a separate pull request. In terms of this one here, I think we should be OK with the local definition, or we can simply just remove the macro and put __attribute__ ((weak, alias(default_handler))) directly in the prototype.

Maybe. No need to remove the macro.

Nothing should be required in the platform code, not even a `CC_CONF_WEAK_ALIAS`. The availability only depends on the toolchain and on the CPU (core) family. Many different MCUs / SoCs / platforms use the same toolchain and CPU, e.g. with GCC ARM. Everything should be auto-detected, and not configured.

How about this in `cc.h`?
```
#if __STDC_VERSION__ >= 199901L
#define PRAGMA(x) _Pragma(#x)
#endif

#if __GNUC__
#define WEAK_ALIAS(alias, ref) __attribute__((__weak__, __alias__(#ref)))
#elif __ICCARM__ || __ICCSTM8__
#define WEAK_ALIAS(alias, ref) PRAGMA(weak alias=ref)
#endif
```

In this way, if `WEAK_ALIAS()` is used where unsupported, the build will simply fail, and all cases supporting it are auto-detected and make it available.

These macros could even be built on the function prototypes if position matters, or to avoid duplicating the function name, but I think this would be too much before we find a case where it's strictly required.